### PR TITLE
Feature/64925 azure functions upgrade to v4 API

### PR DESCRIPTION
**Pull request by GuyHarwood**
Upgrade `@azure/functions` to v4 API
- refactor function bindings from JSON files to inline
- House utility functions in their own folder
- Target function definitions from package.json

### CODE REVIEW

File: func-v4-poc/src/functions/timerTrigger.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { app, InvocationContext, Timer } from "@azure/functions";
+
+export async function timerTrigger(myTimer: Timer, context: InvocationContext): Promise<void> {
+    context.log('Timer function processed request.');
+}
+
+app.timer('timerTrigger', {
+    schedule: '0 */5 * * * *',
```
Comment (jon-shipley) [inline func-v4-poc/src/functions/timerTrigger.ts:8]: Can we get a description of this timer schedule in English please?

File: func-v4-poc/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "target": "es6",
```
Comment (jon-shipley) [inline func-v4-poc/tsconfig.json:4]: We could consider targeting something more modern like es2021 or es2023.

File: tslib/src/functions/check-notifier-batch/index.ts [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
 import config from '../../config'
 import { type ICheckNotificationMessage } from '../../schemas/check-notification-message'
 import { BatchCheckNotifier } from './batch-check-notifier.service'
 import * as RA from 'ramda-adjunct'
-import { type IFunctionTimer } from '../../azure/functions'
 
 const functionName = 'check-notifier-batch'
+
+app.timer('checkNotifierBatch', {
+  schedule: '*/30 * * * * *',
```
Comment (jon-shipley) [inline tslib/src/functions/check-notifier-batch/index.ts:12]: Can we get an English description on this schedule please?

**Review by jon-shipley — COMMENTED** [review]
LGSF 🏅

File: func-v4-poc/src/functions/serviceBusQueueTrigger.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    PartitionKey: 'messages',
+    RowKey: new Date().getTime().toString(),
+    Message: message
+  }])
+}
+
+
+
+
+/* input.generic({
```
[the remark this answers is not in the document]
Comment (GuyHarwood) [reply func-v4-poc/src/functions/serviceBusQueueTrigger.ts:41]: this app will be removed before the PR is completed, as its just a shell that exercises how each trigger type is declared.

File: tslib/src/functions-util/create-user/index.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+async function createUser (context: InvocationContext, req: HttpRequest): Promise<HttpResponseInit> {
+  if (req.body === undefined) {
+    return generateResponse(context, 'Failed', 400, 'Missing body')
+  }
+
+  try {
+    const userApi = new UserApi(context)
+    const newUserInfo = await req.json() as ICreateUserModel
+    const entity = await userApi.create(newUserInfo)
+    context.log(`GUY: entity created... ${JSON.stringify(entity)}`)
```
Comment (jon-shipley) [inline tslib/src/functions-util/create-user/index.ts:60]: debug

File: tslib/src/functions-util/create-user/index.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    return generateResponse(context, 'Success', 201, 'Created', entity)
+  } catch (error) {
+    context.error(`GUY: error caught: ${error}`)
```
Comment (jon-shipley) [inline tslib/src/functions-util/create-user/index.ts:63]: debug

File: tslib/src/functions-util/create-user/user-api.ts [hunk 7]
```diff
@@ -52,17 +59,20 @@ export class UserApi {
         VALUES (@passwordHash, @schoolId, @roleId, @identifier)
     `
     const params = [
-      { name: 'urlSlug', value: schoolUuid, type: TYPES.UniqueIdentifier },
+      { name: 'urlSlug', value: schoolUUID, type: TYPES.UniqueIdentifier },
       { name: 'passwordHash', value: passwordHash, type: TYPES.NVarChar },
       { name: 'roleTitle', value: role.toUpperCase(), type: TYPES.NVarChar },
       { name: 'identifier', value: identifier, type: TYPES.NVarChar(64) }
     ]
+    this.logger.trace(`GUY: inserting user with params: ${JSON.stringify(params)}`)
```
Comment (jon-shipley) [inline tslib/src/functions-util/create-user/user-api.ts:67]: debug

File: tslib/src/functions-util/create-user/user-api.ts [hunk 8]
```diff
@@ -52,17 +59,20 @@ export class UserApi {
...
     await this.sqlService.modify(sql, params)
+    this.logger.trace('GUY: user inserted, fetching from db')
```
Comment (jon-shipley) [inline tslib/src/functions-util/create-user/user-api.ts:69]: debug

File: tslib/src/functions-util/create-user/user-api.ts [hunk 9]
```diff
@@ -52,17 +59,20 @@ export class UserApi {
...
     const data = await this.sqlService.query('SELECT * from mtc_admin.[user] WHERE identifier = @identifier', [{
       name: 'identifier',
       value: identifier,
       type: TYPES.NVarChar
     }])
+    this.logger.trace(`GUY: user fetched from db: ${JSON.stringify(data)}`)
```
Comment (jon-shipley) [inline tslib/src/functions-util/create-user/user-api.ts:75]: debug

**Review by jon-shipley — APPROVED** [review]
LGTM 🏆 

(some debug msgs found)

**Pull request merged**